pypsa/plot.py

```diff
@@ -95,9 +95,14 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
         installed. If None (default) is passed the projection for cartopy
         is set to cartopy.crs.PlateCarree
     bus_colors : dict/pandas.Series
-        Colors for the buses, defaults to "b"
+        Colors for the buses, defaults to "b". If bus_sizes is a pandas.Series
+        with a Multiindex, bus_colors defaults to the network.carriers['color']
+        column.
     bus_sizes : dict/pandas.Series
-        Sizes of bus points, defaults to 10
+        Sizes of bus points, defaults to 10. If a multiindexed Series is passed,
+        the function will draw pies for each bus (first index level) with
+        segments of different color (second index level). Such a Series is ob-
+        tained by e.g. network.generators.groupby(['bus', 'carrier']).p_nom.sum()
```

Comment:
Thanks, that feature wasn't even documented yet.

Reply:
Yes I was surprised as well!